flatnav/util/VisitedSetPool.h

```diff
@@ -170,6 +170,21 @@ class VisitedSetPool {
     _visisted_set_pool.push_back(visited_set);
   }
 
+  void setPoolSize(uint32_t new_pool_size) {
```

Comment:
Added this just to make sure that if we are in the search regime, we don't have more visited set allocations than we need (we need exactly one set). So, when we run `set_num_threads` it's a good idea to deallocate other visited sets.